### PR TITLE
Add Response and change responsibility of Reply.

### PR DESCRIPTION
**Pull request by k-owens**
Response has public properties for application to set.  Reply creates the message to send to the client based off of Response.

### CODE REVIEW

File: Server.Core/IResponseHandler.cs [hunk 1]
```diff
@@ -2,6 +2,6 @@
 {
     public interface IResponseHandler
     {
-        Reply Execute(Request request);
+        Response HandleResponse(Request request);
```
Comment (croese) [inline Server.Core/IResponseHandler.cs]: technically, you're handling a `Request` and the result is a `Response`...

File: Server.Core/Reply.cs [hunk 2]
```diff
@@ -34,6 +31,57 @@ public byte[] ReplyMessage()
             return wholeMessage;
         }
 
+        private byte[] GetBodyMessage()
+        {
+            if (!(_response.Body == null))
```
Comment (croese) [inline Server.Core/Reply.cs]: invert the conditional here so the happy path is not indented

File: Server.Core/Reply.cs [hunk 3]
```diff
@@ -34,6 +31,57 @@ public byte[] ReplyMessage()
...
+                var buffer = new byte[(int)_response.Body.Length];
+                _bodySize = _response.Body.Read(buffer, 0, buffer.Length);
+                return buffer;
+            }
+            return new byte[0];
+        }
+
+        private byte[] CreateHeaders()
+        {
+            if (_bodySize != 0)
```
Comment (croese) [inline Server.Core/Reply.cs]: same conditional comment as above

File: Server.Core/Reply.cs [hunk 4]
```diff
@@ -34,6 +31,57 @@ public byte[] ReplyMessage()
...
+            {
+                if (!(_response.ContentType == null))
+                {
+                    return Encoding.UTF8.GetBytes("Content-Type: " + _response.ContentType + "\r\n" + "Content-Length: " + _bodySize + "\r\n");
```
Comment (croese) [inline Server.Core/Reply.cs]: for this and the other methods that just have static text being encoded into bytes, it'd probably be less work to just deal with strings and then turn everything into bytes right before returning to the client

**Review by croese — COMMENTED** [review]
i like the separation between the Reply and Response, one for the client and one that handles more of the nitty-gritty details

**Comment by k-owens** [comment]
I'll update it based on the suggestions.

**Pull request merged**